CONTRIBUTING.md

```diff
@@ -0,0 +1,75 @@
+
+## Releases
+
+Trust-DNS tries to follow semver versioning semantics. Major versions will not break APIs in a current major revision. If changes are being made to the current `main` branch, double check the current status of the Major release. Until `1.x.x`, all `0.x.x` minor releases are treated as major releases with breaking changes allowed. Releases are performed on an ad-hoc/on-demand basis.
+
+*Maintainers*: If changes are needed to previous releases, then there should exist a `releease/x.x`. If this does not exist, then go to the previous most recent tag (release) and create a new branch at that tag `release/x.x`, for example the branch `release/0.19`:
+
+```shell
+> git fetch origin
+> git checkout v0.19
+> git branch release/0.19
+> git push --set-upstream origin release/0.19
+```
+
+Previous release can fail due to time-lapse, please see **Updating Security Related Tests**.
+
+## Performing a Release, for Maintainers
+
+Releases are somewhat automated. The github action, `publish`, watches for any tags on the project. It then attempts to perform a release of all the libraries, this does not always work, for various reasons.
+
+1. Create a new branch like `git checkout -b prepare-0.20.1`
+2. Update all Cargo.toml files to the new version, `version = 0.20.1`
+3. Update all inter-dependent crates, i.e. trust-dns-resolver to use `trust-dns-proto = 0.20.1`
+4. Update [CHANGELOG.md](CHANGELOG.md) to include all PR's (of consequence) since the previous release
+5. Push to Github, create a PR and merge in `main` or the target release branch.
+6. Go to [Releases](https://github.com/bluejekyll/trust-dns/releases) and `Draft a new release`
+7. Give it a `Tag Version` of `vX.x.x`, e.g. `v0.20.1`, *make sure this is tagging the correct branch, e.g. `main` or `release/0.19`*
+8. Give it a  `Release Title` of something key to the release
+9. Copy and pase the part of the CHANGELOG.md for this release into `Describe this release`
+10. `Publish Release`, this will kick off the publish workflow
+
+After approximately 45 minutes it should be published. This may fail.
+
+**TBD**: add instructions about using Makefile.toml to skip already published crates
+
+## Updating Security Related Tests
+
+TBD: add notes on updating Makefile.toml for OpenSSL version
+TBD: add notes on updating certificates in test directories
+
+## FAQ
+
+- Why are there so few maintainers?
+
+There have not been that many people familiar with DNS internals, networking, security, and Rust that the list of maintainers has been relatively small.
+
+- Will new maintainers be considered?
+
+Yes! There is no formal process, and while generally it's a goal to open up to anyone, we'd ask that you are committed to the goals of an open DNS implementation that anyone can freely use as they see fit. Please reach out on Discord if you'd like to become a maintainer and discuss with us.
```

Comment:
I feel like the normal route would be to at least get a few PRs merged before applying to be a maintainer. If you agree, might be nice to make that explicit?

Reply:
Yeah, I wasn't really sure here, I like that as an additional detail.